solutions/usaco-531.mdx

```diff
@@ -122,3 +122,64 @@ int main()
 </CPPSection>
 
 </LanguageSection>
+
+
+## Prim's Implementation
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+	
+public class Superbull {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("superbull");
+		int N = io.nextInt();
+		// Read in the input array.
+		int[] ids = new int[N];
+		for (int i = 0; i < N; i++) {
+			ids[i] = io.nextInt();
+		}
+
+		long max_cost = 0;
+		boolean[] visited = new boolean[N];
+		int[] score = new int[N];
+
+		// Loop through possible choices for the first team.
+		for (int team1 = 0; team1 < N; team1++) {
+			int team2 = -1;
+
+			// Loop through possible choices of the next team.
+			for (int i = 0; i < N; i++) {
+				if (visited[i]) {
+					continue;
+				}
+				if (team2 == -1 || (score[i] > score[team2])) {
+					team2 = i;
+				}
+			}
+
+			// Pair team1 and team2.
+			max_cost += score[team2];
+			visited[team2] = true;
+
+			// Re-evaluate all pair scores.
+			for (int i = 0; i < N; i++) {
+				score[i] = Integer.max(score[i], (ids[team2] ^ ids[i]));
+			}
+		}
+		pw.println(max_cost);
+		pw.close();
```

Comment:
io?

Reply:
was this tested?

Reply:
fixed

Reply:
it works
[image](https://ibb.co/wgLHRdn)
[code](https://pastebin.com/fErXW15Y)